code/utils/Random.cpp

```diff
@@ -0,0 +1,71 @@
+#include "Random.h"
+
+#include <limits>
+#include <random>
+
+#include <cstdint>
+
+#include "globalincs/pstypes.h"
+
+namespace util {
+namespace {
+template <typename RngType>
+class RandomImpl {
+public:
+void seed(const unsigned int val) {
+	static_assert(RngType::min() == 0, "RNG min must be 0");
+	static_assert(RngType::max() == std::numeric_limits<uint32_t>::max(), "RNG max must be INT32_MAX");
+
+	m_rng.seed(val);
+}
+
+int next()
+{
+	return m_rng() & Random::MAX_VALUE;
+}
+
+void advance(unsigned long long distance)
+{
+	m_rng.discard(distance);
+}
+
+private:
+	RngType m_rng;
+};
+
+RandomImpl<std::mt19937> SCP_rng;
+} // namespace
+
+Random::Random() = default;
+
+void Random::seed(unsigned int val)
+{
+	Assert(val > 0);
+	SCP_rng.seed(val);
+}
+
+int Random::next()
+{
+	return SCP_rng.next();
+}
+
+int Random::next(int modulus)
+{
+	Assert(modulus > 0);
+
+	return next() % modulus;
+}
+
+int Random::next(int low, int high)
+{
+	const int range = high - low + 1;
+	Assert(range > 0);
+
+	return low + (SCP_rng.next() % range);
+}
```

Comment:
This could use a comment to make it explicit that this is an inclusive range that can return both `low` and `high`, particularly as the definition immediately above is _exclusive_ at the upper bound and cannot return `modulus`.